rust/noosphere-gateway/src/worker/name_system.rs

```diff
@@ -147,115 +147,132 @@ where
         }
     });
 
+    let ipfs_api = configuration.ipfs_api.clone();
     while let Some(job) = receiver.recv().await {
-        let ipfs_api_url = configuration.ipfs_api.clone();
-        let run_job = with_client.invoke(|client| async move {
-            debug!("Running {}", job);
-            match job {
-                NameSystemJob::Publish { record, .. } => {
-                    client.publish(NsRecord::from_str(&record)?).await?;
+        if let Err(error) = process_job(job, &mut with_client, &ipfs_api).await {
+            warn!("Error processing NS job: {}", error);
+        }
+    }
+    Ok(())
+}
+
+async fn process_job<C, K, S, I, O, F>(
+    job: NameSystemJob<C>,
+    with_client: &mut TryOrReset<I, O, F>,
+    ipfs_api: &Url,
+) -> Result<()>
+where
+    C: HasMutableSphereContext<K, S>,
+    K: KeyMaterial + Clone + 'static,
+    S: Storage + 'static,
+    I: Fn() -> F,
+    O: NameResolver + 'static,
+    F: Future<Output = Result<O, anyhow::Error>>,
+{
+    let run_job = with_client.invoke(|client| async move {
+        debug!("Running {}", job);
+        match job {
+            NameSystemJob::Publish { record, .. } => {
+                let link_record = NsRecord::from_str(&record)?;
+                if link_record.has_publishable_timeframe() {
```

Comment:
This file is noisy due to restructuring processing: adding a publishable timeframe check here